File: app/src/main/java/com/tistory/s1s1s1/inu_contact/RetroAsync.java
package com.tistory.s1s1s1.inu_contact;

import android.app.ProgressDialog;
import android.content.Context;
import android.os.AsyncTask;
import android.support.v7.widget.DefaultItemAnimator;

import com.apkfuns.xprogressdialog.XProgressDialog;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.tistory.s1s1s1.inu_contact.RecyclerView.ContactAdapter;

import java.util.ArrayList;

import retrofit2.Call;

import static com.tistory.s1s1s1.inu_contact.MainActivity.actionbar_et_search;
import static com.tistory.s1s1s1.inu_contact.MainActivity.actionbar_tv_title;
import static com.tistory.s1s1s1.inu_contact.MainActivity.main_rv;
import static com.tistory.s1s1s1.inu_contact.MainActivity.rv_level;


/**
 * Created by seonilkim on 2017. 3. 23..
 */

public class RetroAsync extends AsyncTask<Void, Void, Void> {

    private Context context;
    private Singleton singleton;
    private XProgressDialog xProgressDialog = null;
    private JsonArray result = null;
    private DBHelper dbHelper;

    public RetroAsync(Context context, XProgressDialog xProgressDialog, JsonArray jsonArray, DBHelper dbHelper) {
        this.context = context;
        this.xProgressDialog = xProgressDialog;
        this.result = jsonArray;
        this.dbHelper = dbHelper;
    }

    @Override
    protected void onPreExecute() {
        if(singleton == null) singleton.getInstance(context);
    }

    @Override
    protected Void doInBackground(Void... voids) {
        JsonObject object;
        for(int i=0;i<result.size();i++){
            object = result.get(i).getAsJsonObject();
            dbHelper.insert(object);
        }

        return null;
    }

    @Override
    protected void onPostExecute(Void aVoid) {
        main_rv.setAdapter(new ContactAdapter(context, dbHelper.getPart()));
        main_rv.setItemAnimator(new DefaultItemAnimator());
        actionbar_tv_title.setText(R.string.app_name);
        rv_level = 0;
        if (actionbar_et_search.isFocused()) {
            actionbar_et_search.clearFocus();
        }
        xProgressDialog.dismiss();
    }
}

File: app/src/main/java/com/tistory/s1s1s1/inu_contact/Network/RetroAsync.kt
package com.tistory.s1s1s1.inu_contact.Network

import android.os.AsyncTask
import android.support.v4.app.Fragment
import android.util.Log
import com.google.gson.JsonArray
import com.google.gson.JsonObject
import com.tistory.s1s1s1.inu_contact.Fragment.GroupFragment
import com.tistory.s1s1s1.inu_contact.LOG_TAG
import com.tistory.s1s1s1.inu_contact.RecyclerView.ContactAdapter
import com.tistory.s1s1s1.inu_contact.Util.DBHelper

class RetroAsync
(
//                        private val xProgressDialog: MyProgress?,
        private val fragment : Fragment,
        private val result: JsonArray,
        private val dbHelper: DBHelper) : AsyncTask<Unit, Unit, Unit>() {
    override fun onPreExecute() {
        super.onPreExecute()
        Log.d(LOG_TAG, "DB 입력 시작")
    }

    override fun doInBackground(vararg voids: Unit): Unit? {
        var json: JsonObject
        for (i in 0 until result.size()) {
            json = result.get(i).asJsonObject
            dbHelper.insert(json)
        }
        return null
    }

    override fun onPostExecute(aVoid: Unit?) {
        if(fragment is GroupFragment){
//            Log.d(LOG_TAG, "DB 입력 종료" + dbHelper.part.toString())
            fragment.mRV!!.adapter = ContactAdapter(dbHelper.part)
//            actionbar_tv_title.setText(R.string.app_name)
            fragment.mRV!!.adapter.notifyDataSetChanged()

//            if (actionbar_et_search.isFocused) {
//                actionbar_et_search.clearFocus()
//            }
        }
//        fragment..adapter = ContactAdapter(dbHelper.part)
//
//        xProgressDialog!!.dismiss()
    }
}


